ngraph/test/preprocess.cpp

```
@@ -208,32 +221,87 @@ TEST(pre_post_process, test_2_inputs_basic) {
+                         .input(InputInfo()
+                                    .tensor(InputTensorInfo().set_layout("N?HW"))
+                                    .preprocess(PreProcessSteps().scale({0.1f, 0.2f, 0.3f})))
+                         .build(f),
+                 ov::AssertFailure);
+}
+
+TEST(pre_post_process, scale_vector_channels_out_of_range) {
+    auto f = create_simple_function(element::f32, Shape{1, 3, 224, 224});
+    ASSERT_EQ(f->get_output_element_type(0), element::f32);
+    ASSERT_THROW(f = PrePostProcessor()
+                         .input(InputInfo()
+                                    .tensor(InputTensorInfo().set_layout("0123C"))
+                                    .preprocess(PreProcessSteps().scale({0.1f, 0.2f, 0.3f})))
+                         .build(f),
+                 ov::AssertFailure);
 }
 
 TEST(pre_post_process, mean_vector_no_layout) {
     auto f = create_simple_function(element::f32, PartialShape{Dimension::dynamic(), 3, 224, 224});
     ASSERT_EQ(f->get_output_element_type(0), element::f32);
-    ASSERT_ANY_THROW(
-        f = PrePostProcessor().input(InputInfo().preprocess(PreProcessSteps().mean({0.1f, 0.2f, 0.3f}))).build(f));
+    ASSERT_THROW(
+        f = PrePostProcessor().input(InputInfo().preprocess(PreProcessSteps().mean({0.1f, 0.2f, 0.3f}))).build(f),
+        ov::AssertFailure);
+}
+
+TEST(pre_post_process, mean_vector_dynamic_channels_shape) {
+    auto f = create_simple_function(
+        element::f32,
+        PartialShape{Dimension::dynamic(), Dimension::dynamic(), Dimension::dynamic(), Dimension::dynamic()});
+    ASSERT_EQ(f->get_output_element_type(0), element::f32);
+    ASSERT_THROW(f = PrePostProcessor()
+                         .input(InputInfo()
+                                    .tensor(InputTensorInfo().set_layout("NCHW"))
+                                    .preprocess(PreProcessSteps().mean({0.1f, 0.2f, 0.3f})))
+                         .build(f),
+                 ov::AssertFailure);
```

@jane-intel is it valid case from `subtract` operation point of view? In a theory if we reshape such network to `[X 3 Y Z]` it's valid. While for `[X 4 Y Z]` we should throw an error on `reshape` stage

---

Fixed, test is corrected. Now it allows such situation and will throw on evaluation stage if user's tensor mismatches with mean/scale